File: implementation/scipamato/public/web/src/main/java/ch/difty/scipamato/publ/misc/LocaleExtractor.java
package ch.difty.scipamato.publ.misc;

import java.util.Locale;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * Implementations of this interface have the ability
 * of parsing some kind of text value to identify and
 * return the Locale.
 */
@FunctionalInterface
public interface LocaleExtractor {

    /**
     * Extracts the {@link Locale} from given textual input.
     * If the input cannot be parsed correctly, the
     * default locale is returned.
     *
     * @param input
     *     the input value containing the locale
     * @return the locale encoded in the input.
     *     Is never null.
     */
    @NotNull
    Locale extractLocaleFrom(@Nullable String input);
}

File: public/public-web/src/main/kotlin/ch/difty/scipamato/publ/misc/LocaleExtractor.kt
package ch.difty.scipamato.publ.misc

import java.util.Locale

/**
 * Implementations of this interface have the ability of parsing some kind of text value
 * to identify and return the Locale.
 */
fun interface LocaleExtractor {

    /**
     * Extracts the [Locale] from given textual [input].
     * If the input cannot be parsed correctly, the
     * default locale is returned.
     */
    fun extractLocaleFrom(input: String?): Locale
}
